src/labelle/lib/devices/device_config.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import logging
+from typing import Tuple
+
+LOG = logging.getLogger(__name__)
+
+
+class DeviceConfig:
+    """Configuration object for the capabilities of a label printer."""
+
+    name: str
+    """Name of this device"""
+
+    deviceIDs: list[int]
+    """List of USB Device ID's this device can identify as"""
+
+    printHeadSizePixels: int
+    """Size of the print head in pixels (use calibration routine to determine)"""
+
+    supportedTapeSizes: list[int]
+    """List of supported tape sizes in mm"""
+
+    marginsPerTape: dict[int, Tuple[int, int]]
+    """
+    Dictonary of print margins per tape size
+    Entry: TapeSizeMM : (topMarginInPx, bottomMarginInPx)
+    Use the calibration routine to determine these for each tape size
+    """
+
+    def __init__(
+        self,
+        name: str,
+        deviceIDs: list[int],
+        printHeadSizePixels: int,
+        supportedTapeSizes: list[int],
+        marginsPerTape: dict[int, Tuple[int, int]],
+    ):
+        """Initialize a Labeler config object."""
+        self.name = name
+        self.deviceIDs = deviceIDs
+        self.printHeadSizePixels = printHeadSizePixels
+        self.supportedTapeSizes = supportedTapeSizes
+        self.marginsPerTape = marginsPerTape
```

Comment:
Maybe define the class as `@dataclass`. https://docs.python.org/3/library/dataclasses.html

Reply:
Personally, I'd probably use a [`NamedTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple) which is simpler, but either should work.

Reply:
I'm not sure of the pro's and con's of both. But I think dataclass is easier to convert to?
I'll give it a shot

Reply:
Made dataclass